bin/dropshot-apis/src/main.rs

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::process::ExitCode;
+
+use anyhow::Context;
+use camino::Utf8PathBuf;
+use clap::Parser;
+use dropshot_api_manager::{Environment, ManagedApiConfig, ManagedApis};
+use dropshot_api_manager_types::{ManagedApiMetadata, Versions};
+use propolis_server_api::*;
+
+pub fn environment() -> anyhow::Result<Environment> {
+    // The workspace root is two levels up from this crate's directory.
+    let workspace_root = Utf8PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .to_path_buf();
+    let env = Environment::new(
+        // This is the command used to run the OpenAPI manager.
+        "cargo xtask openapi".to_owned(),
+        workspace_root,
+        // This is the location within the workspace root where the OpenAPI
+        // documents are stored.
+        "openapi".into(),
+    )?
+    .with_default_git_branch("origin/master".to_owned());
+    Ok(env)
+}
+
+/// The list of APIs managed by the OpenAPI manager.
+pub fn all_apis() -> anyhow::Result<ManagedApis> {
+    let apis = vec![ManagedApiConfig {
+        ident: "propolis-server",
+        versions: Versions::Lockstep { version: semver::Version::new(0, 0, 1) },
```

Comment:
and presumably in the future we'll move this to `Versioned` with the latest version and whatever the previous release's API was, right? (i don't think this was explicitly written out for APIs outside Omicron before, and being able to write that is part of the motivation here?)

Reply:
Yes, making these APIs versioned (as part of self-service update) is the overall motivation for the series of changes I've been making to all of these repos. See https://github.com/oxidecomputer/omicron/issues/8922 for overall status tracking.